src/rockstor/storageadmin/views/pool.py

```diff
@@ -211,6 +225,39 @@ def _remount(cls, request, pool):
             handle_exception(Exception(e_msg), request)
         return Response(PoolInfoSerializer(pool).data)
 
+    @classmethod
+    def _quotas(cls, request, pool):
+        new_quota_state = cls._validate_new_quota_state(request)
+        # If no change from current pool quota state then do nothing
+        current_state = 'Enabled'
+        if not pool.quotas_enabled:
+            current_state = 'Disabled'
+        if new_quota_state == current_state:
+            return Response()
+        try:
+            if new_quota_state == 'Enabled':
+                # Current issue with requiring enable to be executed twice !!!
+                # As of 4.12.4-1.el7.elrepo.x86_64
+                # this avoids "ERROR: quota rescan failed: Invalid argument"
+                # when attempting a rescan.
+                # Look similar to https://patchwork.kernel.org/patch/9928635/
+                enable_quota(pool)
+                enable_quota(pool)
```

Comment:
It may be useful to open an issue to keep track of future refactoring of this logic.

Reply:
Agreed. I have just created what I think should cover it:
"re-address double enable_quota() call requirement" #1905